BitwardenShared/Core/Platform/Services/StateService.swift

```diff
@@ -1513,11 +1522,13 @@ actor DefaultStateService: StateService { // swiftlint:disable:this type_body_le
     func setAccountSetupAutofill(_ autofillSetup: AccountSetupProgress?, userId: String?) async throws {
         let userId = try userId ?? getActiveAccountUserId()
         appSettingsStore.setAccountSetupAutofill(autofillSetup, userId: userId)
+        try await updateSettingsBadgePublisher(userId: userId)
     }
 
     func setAccountSetupVaultUnlock(_ vaultUnlockSetup: AccountSetupProgress?, userId: String?) async throws {
         let userId = try userId ?? getActiveAccountUserId()
         appSettingsStore.setAccountSetupVaultUnlock(vaultUnlockSetup, userId: userId)
+        try await updateSettingsBadgePublisher(userId: userId)
```

Comment:
🤔 @matt-livefront these methods are currently called in `loginWithMasterPassword` so potentially if `updateSettingsBadgePublisher` throws then `loginWithMasterPassword` would fail which is not correct IMO. The badge update is a secondary effect but that shouldn't block the flow of logging in.
I think the `updateSettingsBadgePublisher` should not throw and just wrap the code in a `do...catch` and log any error to the `errorReporter` so it doesn't affect any caller flows.

Reply:
Thanks, great point! I have a fix here: https://github.com/bitwarden/ios/pull/988.